paymentapi/src/main/java/fi/hel/verkkokauppa/payment/service/PaymentMethodService.java

```diff
@@ -0,0 +1,233 @@
+
+		if (filter.equals(PaymentMethodFilter.ORDER_TYPE)) {
+			values.put(OrderType.SUBSCRIPTION, Collections.singletonList(PaymentType.CREDIT_CARDS));
+		}
+
+		return values;
+	}
+
+	public List<PaymentMethodDto> getAllPaymentMethods() {
+		Iterable<fi.hel.verkkokauppa.payment.model.PaymentMethod> paymentMethods = paymentMethodRepository.findAll();
+
+		List<PaymentMethodDto> paymentMethodDtos = StreamSupport.stream(paymentMethods.spliterator(), false)
+				.map(paymentMethod -> mapper.convertValue(paymentMethod, PaymentMethodDto.class))
+				.collect(Collectors.toList());
+		return paymentMethodDtos;
+	}
+
+	public PaymentMethodDto getPaymenMethodByCode(String code) {
+		fi.hel.verkkokauppa.payment.model.PaymentMethod exsistingMethod = paymentMethodRepository.findByCode(code).stream()
+				.filter(paymentMethod -> paymentMethod.getCode().equals(code))
+				.findFirst()
+				.orElseThrow(() -> new CommonApiException(
+						HttpStatus.NOT_FOUND,
+						new Error("payment-method-not-found", "payment method with code [" + code + "] not found")
+				));
+		PaymentMethodDto dto = mapper.convertValue(exsistingMethod, PaymentMethodDto.class);
+		return dto;
+	}
+
+	public PaymentMethodDto createNewPaymentMethod(PaymentMethodDto dto) {
+		paymentMethodRepository.findByCode(dto.getCode()).stream()
+				.filter(paymentMethod -> paymentMethod.getCode().equals(dto.getCode()))
+				.findFirst()
+				.ifPresent(paymentMethod -> {
+					throw new CommonApiException(
+							HttpStatus.CONFLICT,
+							new Error("payment-method-already-exists", "payment method with code [" + paymentMethod.getCode() + "] already exists")
+					);
+				});
+
+		fi.hel.verkkokauppa.payment.model.PaymentMethod paymentMethod = mapper.convertValue(dto, fi.hel.verkkokauppa.payment.model.PaymentMethod.class);
+		fi.hel.verkkokauppa.payment.model.PaymentMethod saved = paymentMethodRepository.save(paymentMethod);
+
+		return new PaymentMethodDto(saved.getName(), saved.getCode(), saved.getGroup(), saved.getImg(), saved.getGateway());
+	}
+
+	public PaymentMethodDto updatePaymentMethod(String code, PaymentMethodDto dto) {
+		fi.hel.verkkokauppa.payment.model.PaymentMethod paymentMethodToUpdate  = paymentMethodRepository.findByCode(code).stream()
+				.filter(paymentMethod -> paymentMethod.getCode().equals(code))
+				.findFirst()
+				.orElseThrow(() -> new CommonApiException(
+						HttpStatus.NOT_FOUND,
+						new Error("payment-method-not-found", "payment method with code [" + code + "] not found")
+				));
+		paymentMethodToUpdate.setName(dto.getName());
+		paymentMethodToUpdate.setCode(dto.getCode());
+		paymentMethodToUpdate.setGroup(dto.getGroup());
+		paymentMethodToUpdate.setImg(dto.getImg());
+		paymentMethodToUpdate.setGateway(dto.getGateway());
+
+		fi.hel.verkkokauppa.payment.model.PaymentMethod saved = paymentMethodRepository.save(paymentMethodToUpdate);
+
+		return new PaymentMethodDto(saved.getName(), saved.getCode(), saved.getGroup(), saved.getImg(), saved.getGateway());
```

Comment:
You should use object mapper, no need for constructor for dto.
Like:
return mapper.convertValue(saved, PaymentMethodDto.class)